### ISSUE TITLE
Managing ops-usage documentation for flimj-ops

### ISSUE THREAD

**Issue by JenuC** [flimlib/flimlib-javadoc#1]
@ctrueden : If I want to add some documentation, would it be helpful to add it in this repository or as comments in the flimj-ops/flimlib repo? or rather, how can I make the help-section for groovy functions a bit more descriptive?

**Comment by JenuC** [comment]
I guess the right place would in the [sites](https://github.com/flimlib/flimlib.github.io). Is it? But then I cannot refer to functions and parameters associated with it.  Maybe I am looking for a parallel to the slim-curve doxygen pages.

**Comment by ctrueden** [comment]
@JenuC Ah! You use the Doxygen pages? @paulbarber and I discussed that awhile back, and we concluded they weren't necessary to keep, but if you find them useful, I could look at reviving them for FLIMLib. What do you think, @paulbarber?

As for documentation in general:

* FLIMLib = http://flimlib.github.io/
* FLIMJ = https://imagej.net/FLIMJ

With FLIMJ docs linked from the FLIMLib site.

The Groovy functions should be documented on the FLIMJ side I think, since that's really just using the ImageJ-specific API. @Xanthorapedia @paulbarber Do you agree?

So for now, how about making a `Groovy.md` file with the information? Then we can merge it into FLIMJ.md on imagej.github.io (the new home of imagej.net) once we've enabled write-access there.

@JenuC On the other hand, if what you're trying to do is improve the documentation of the various Java-side FLIMJ functions, then the best place to do that would be in the Javadoc—so https://github.com/flimlib/flimj-ops and https://github.com/flimlib/flimj-ui repositories. Can you give an example of what you have in mind documentation-wise?

**Comment by JenuC** [comment]
@ctrueden : doxygen :grimacing: , not quite! I just learned about doxygen only yesterday.  But I liked the way it had all the function declarations and the formal parameters in it. I think the solution is to add in the https://imagej.net/FLIMJ (after current migration).  For now, I'll work with @Xanthorapedia to put together a groovy.md in the flimj page that can be later linked to the flimj.io or flimj-javadocs, or similar. Thanks!

**Comment by JenuC** [comment]
added a document: https://github.com/flimlib/flimj-ops/blob/master/groovy.md
close the issue?

**Comment by paulbarber** [comment]
Agreed. The Doxygen was for the c code api to help with making FLIMJ. Now we need docs for all the different ways to use FLIMJ, including detailed api docs that Jenu requires now and I will require later 😉

**Comment by JenuC** [comment]
@ctrueden : I want to add some descriptions to the flimj-ops functions as you suggested. I would start transferring some of the c-level documentation to the ops for understanding the parameters. Is there an available example of _javadoc documentation for ops_ that I can use as a reference? I am imagining a documentation that would be helpful in ImageJ scripting window and/or groovy notebooks, primarily for users like me. thanks in advance! :bow:

**Comment by JenuC** [comment]
After looking at some other ops-functions, I think my solution is to add "Method Detail" section to the flimj-ops functions, so that 
[flimj-ops javadocs](https://javadoc.scijava.org/FLIMLib/flimlib/flimj/package-summary.html) page would reflect the detail. I don't know how to add the method detail/summary yet, but that is my current direction now. 😄  but I may be biting off more than what I can chew :sweat_smile: 
```java
/**
* Global Fit fits all the pixels to generate a spatial map of contributions of two or more known species. 
*This is different from Lifetime invariant fitting or Global binning that generates the decay profile
* integrated over all the pixels as a single decay profile that can be fitted to a decay model. 
* Refer: Barber PR (2005) Global and pixel kinetic data analysis for FRET detection 
* by multi-photon time-domain FLIM. Proc SPIE: 171–181. doi:10.1117/12.590510
* Please see the {@link https://www.sciencedirect.com/science/article/pii/S0006349504737510} 
* @author ?
* 
*/

public GlobalFitWorker(FitParams<I> params, FitResults results, OpEnvironment ops) {
		super(params, results, ops);
	}
	
	/**
	 * <p> Global Fit worker 
	 * <a href="https://www.sciencedirect.com/science/article/pii/S0006349504737510">Link to the article!</a>
	 * </p>
	 * @param ?
	 * @return ?
	 * 
	 */
	 
	 // trans data and fitted parameters for each trans
		final float[][] trans = new float[nTrans][nDataTotal];
```
